File: api/src/Database/Models/Member.cs
namespace Kratos.Api.Database.Models;

public class Member
{
    public long Id { get; set; }
    public string FullName { get; set; } = null!;
    public string Bio { get; set; } = null!;
    public string DisplayPictureFileName { get; set; } = null!;
    public string Position { get; set; } = null!;

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: client/web/src/features/members/types/member.ts
import type { ActionType } from "@/constants/actionType";
import type { Column } from "@/types/column";

export type Member = {
  id: number;
  fullName: string;
  bio: string;
  position: string;
  displayPictureFileName: string;
};

export type AddMemberRequest = {
  fullName: string;
  bio: string;
  position: string;
  displayPicture: File;
};

export type EditMemberRequest = {
  id: number;
  fullName: string;
  bio: string;
  position: string;
  displayPicture?: File;
};

export type MemberActions = {
  actionName: ActionType.Edit | ActionType.Delete;
  actionFn: () => EditMemberRequest;
};

export function createMemberMappings(): Column[] {
  return [
    {
      header: "Full Name",
      accessorKey: "fullName",
    },
    {
      header: "Bio",
      accessorKey: "bio",
    },
  ];
}
